File: app/src/main/java/com/fphoenixcorneae/viewpagertransformer/FlipVerticalTransformer.java
package com.fphoenixcorneae.viewpagertransformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * Created on 2017/6/30.
 */

public class FlipVerticalTransformer extends BaseTransformer {

    @Override
    protected void onTransform(View page, float position) {
        float rotation = -180.0F * position;
        ViewHelper.setAlpha(page, rotation > 90.0F || rotation < -90.0F ? 0F : 1F);
        ViewHelper.setPivotX(page, page.getWidth() * 0.5F);
        ViewHelper.setPivotY(page, page.getHeight() * 0.5F);
        ViewHelper.setRotationX(page, rotation);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/view_pager_transformer/FlipVerticalTransformer.kt
package com.fphoenixcorneae.view_pager_transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper

/**
 * @desc 垂直快速翻转的转换效果
 * @date 2017-06-30
 */
class FlipVerticalTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        val rotation = -180.0f * position
        ViewHelper.setAlpha(page, if (rotation > 90.0f || rotation < -90.0f) 0f else 1f)
        ViewHelper.setPivotX(page, page.width * 0.5f)
        ViewHelper.setPivotY(page, page.height * 0.5f)
        ViewHelper.setRotationX(page, rotation)
    }
}